### PR TITLE
Improve error message when using a restricted pin.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: lib/gpio/pin.toit [hunk 1]
```diff
@@ -391,7 +393,10 @@ gpio-init_:
   #primitive.gpio.init
 
 gpio-use_ resource-group num allow-restricted:
-  #primitive.gpio.use
+  #primitive.gpio.use:
+    if it == RESTRICTED-PIN-ERROR_:
```
Comment (kasperl) [inline lib/gpio/pin.toit:397]: What is going on with this "RESTRICTED" <-> "RESTRICTED_PIN" dance?

File: src/resources/gpio_esp32.cc [hunk 2]
```diff
@@ -38,6 +38,8 @@
 
 namespace toit {
 
+static const char* const RESTRICTED_PIN_ERROR = "RESTRICTED";
```
Comment (kasperl) [inline src/resources/gpio_esp32.cc:41]: Can we use one of the predefined / preallocated ones instead?

File: lib/gpio/pin.toit (hunk 1 above)
Comment (floitsch) [reply lib/gpio/pin.toit]: removed.
done.

File: src/resources/gpio_esp32.cc (hunk 2 above)
Comment (floitsch) [reply src/resources/gpio_esp32.cc]: Switched to "PERMISSION_DENIED".
done.

File: src/primitive.h [hunk 3]
```diff
@@ -1225,6 +1225,7 @@ class Primitive {
   static HeapObject* mark_as_error(HeapObject* object) { return object->mark(); }
   static Object* unmark_from_error(Program* program, Object* object);
   static Object* os_error(int error, Process* process);
+  static Object* error(const char* error, Process* process);
```
Comment (kasperl) [inline src/primitive.h:1228]: This is not needed anymore.

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: src/primitive.h (hunk 3 above)
Comment (floitsch) [reply src/primitive.h]: done.

**Pull request merged**